packages/calcite-components/src/components/block/block.e2e.ts

```diff
@@ -323,6 +323,29 @@ describe("calcite-block", () => {
       const actionAssignedSlot = await page.$eval("calcite-action", (action) => action.assignedSlot.name);
       expect(actionAssignedSlot).toBe(SLOTS.headerMenuActions);
     });
+
+    it("applies correct header spacing when heading or description properties are present", async () => {
```

Comment:
This is fine for now until we add extra checks to ensure styles use reflected props. After that, we should rely on screenshot tests. From the assertion perspective, it is not guaranteed that changes in the CSS class will be detected by this test, meaning its implementation could change without causing a test failure. cc @driskull

Reply:
@jcfranco Should I create a follow up issue to add a screenshot test?

Reply:
@jcfranco I created this follow up [issue](https://github.com/Esri/calcite-design-system/issues/10084) to remove this E2E test once the reflected prop checks are in place.